runtime/src/main/java/org/corfudb/util/MetricsUtils.java

```diff
@@ -31,43 +26,37 @@
     private static final MetricSet metricsJVMMem = new MemoryUsageGaugeSet();
     private static final MetricSet metricsJVMThread = new ThreadStatesGaugeSet();
 
-    private static Properties metricsProperties = new Properties();
+    private static int interval;
     @Getter
-    private static boolean metricsCollectionEnabled = true;
+    private static boolean metricsCollectionEnabled = false;
     private static boolean metricsReportingEnabled = false;
     private static String mpTrigger = "filter-trigger"; // internal use only
 
     /**
-     * Load a metrics properties file.
+     * Load metrics properties.
      *
      * <p>The expected properties in this properties file are:
      * <ul>
-     * <li> collection-enabled: Boolean for whether metrics collection is enabled.
-     * <li> reporting-enabled: Boolean for whether CSV output will be generated.
-     * <li> directory: String for the path to the CSV output subdirectory
-     * <li> nterval: Long for the reporting interval for CSV output
+     * <li> interval: Integer taken from vm metricsInterval property
+     * for the reporting interval for CSV output
      * </ul>
      *
-     * <p>For each reporting interval, this function will be
-     * called to re-parse the properties file and to
-     * re-evaluate the value of 'collection-enabled' and
-     * 'reporting-enabled'.  Changes to any other property
-     * in this file will be ignored.
+     * <p>This function will be called to set the  value of interval for
+     * reporting. Absence of an Integer interval value or a non-positive
+     * value will led to disabling metrics collection and reporting.
      */
-    private static void loadPropertiesFile() {
-        String propPath;
-
-        if ((propPath = System.getenv("METRICS_PROPERTIES")) != null) {
-            try (FileInputStream is = new FileInputStream(propPath)) {
-                metricsProperties.load(is);
-                metricsCollectionEnabled = Boolean.valueOf((String) metricsProperties
-                        .get("collection-enabled"));
-                metricsReportingEnabled = Boolean.valueOf((String) metricsProperties
-                        .get("reporting-enabled"));
-            } catch (Exception e) {
-                log.error("Error processing METRICS_PROPERTIES {}: {}", propPath,
-                        e.toString());
-            }
+    private static void loadVmProperties() {
+        try {
+            interval = Integer.valueOf(System.getProperty("metricsInterval"));
+        } catch (NumberFormatException e) {
+            // interval defaults to -1 as disabled when the value cannot be
+            // extracted from VM property
+            interval = -1;
+        }
+
+        if (interval > 0) {
```

Comment:
what's the difference between metrics collection/reporting?

I think an application would want collection without reporting, or collection and reporting, but a +/- interval only enables/disables both options. Maybe use 0 for collection, but not reporting, neg-interval would be to disable everything and a pos-number would mean a reporting interval.

Reply:
Done.
Collection without reporting might have limited value considering that dropwizard limited expressiveness at debugging at each point of time. However, it's a good idea to have that ability if application wants to have this ability for introducing a manual reporting. Added interval=0 for collection without reporting.